include/pybind11/eigen.h

```diff
@@ -641,6 +643,343 @@ struct type_caster<Type, enable_if_t<is_eigen_other<Type>::value>> {
     using cast_op_type = Type;
 };
 
+template <int Options>
+struct eigen_to_numpy {};
+
+template <>
+struct eigen_to_numpy<Eigen::ColMajor> {
+    static constexpr int flag = array::f_style;
+};
+
+template <>
+struct eigen_to_numpy<Eigen::RowMajor> {
+    static constexpr int flag = array::c_style;
+};
+
+template <class T>
+struct eigen_helper {};
+
+template <typename E, int dim, int O>
+struct eigen_helper<Eigen::Tensor<E, dim, O>> {
+    using Type = Eigen::Tensor<E, dim, O>;
+    using ConstType = Eigen::Tensor<const E, dim, O>;
+
+    using Element = E;
+
+    static constexpr int N = dim;
```

Comment:
What is the point of these constexpr, just reuse the template parameters and give them longer names if you need to. (N is not that much shorter than dim and is of the same type).

Reply:
Removed as requested.